x-pack/plugins/observability_solution/investigate_app/server/routes/get_global_investigate_app_server_route_repository.ts

```diff
@@ -334,12 +334,17 @@ const getEventsRoute = createInvestigateAppServerRoute({
     const alertsClient: AlertsClient = await getAlertsClient({ plugins, request });
     const events: GetEventsResponse = [];
 
-    if (annotationsClient) {
+    const includeAllEventTypes = !params?.query?.eventTypes || params.query.eventTypes.length === 0;
+
+    if (
+      annotationsClient &&
+      (includeAllEventTypes || params?.query?.eventTypes?.includes('annotation'))
```

Comment:
I wonder how robust is this check `params?.query?.eventTypes?.includes('annotation'))`? Do we have another way to check this?

Reply:
Zod ensures that if `eventTypes` is present on the query parameters, then it is an array of event type (enum).

I could write the same check as `params && params.query && params.query.eventTypes && params.queryEventTypes.includes('annotation')` but I would be reinventing optional chaining. I don't think it's more readable this way, but it's definitely more verbose 😅

Reply:
@fkanout Do you want me to address this somehow?

Reply:
Ah yes good point. I forgot to parse the eventTypes:  https://github.com/elastic/kibana/pull/202829/commits/ac24d916fa2b5e2a0d5c6818162ecb6c584101fc